src/reconcile/rec_col.c

```diff
@@ -770,9 +771,18 @@ __wt_rec_col_fix(
         __bit_setv(r->first_free, recno - curstartrecno, btree->bitcnt, val);
 
         /* Write the time window. */
-        if (!WT_TIME_WINDOW_IS_EMPTY(&upd_select.tw))
+        if (!WT_TIME_WINDOW_IS_EMPTY(&upd_select.tw)) {
+            /*
+             * When an out-of-order or mixed-mode tombstone is getting written to disk, remove any
+             * historical versions that are greater in the history store for this key.
+             */
+            if (upd_select.ooo_tombstone && r->hs_clear_on_tombstone)
+                WT_ERR(__wt_rec_hs_clear_on_tombstone(
+                  session, r, upd_select.tw.durable_stop_ts, recno, NULL, true));
```

Comment:
Here we should clear the entries from the history store and should not reinsert them, so I think the last parameter should be false instead of true. The parameters are different when compared with line 761.

Reply:
No. The change is expected. We do not want to reinsert the existing HS entries when the data store key is removed, but we want to reinsert them when the tombstone is written to the data store. That last variable represents whether the reinsert is required or not.